src/network/networkd-ndisc.c

```diff
@@ -192,12 +260,62 @@ static int ndisc_router_process_default(Link *link, sd_ndisc_router *rt) {
         return 0;
 }
 
+static int ndisc_router_generate_address(Link *link, unsigned prefixlen, uint32_t lifetime_preferred, Address *address) {
+        bool prefix_stable = false;
+        struct in6_addr addr;
+        IPv6Token *j;
+        Iterator i;
+        int r;
+
+        assert(address);
+        assert(link);
+
+        addr = address->in_addr.in6;
+        ORDERED_HASHMAP_FOREACH(j, link->network->ipv6_tokens, i) {
+                if (j->address_generation_type == IPV6_TOKEN_ADDRESS_GENERATION_PREFIXSTABLE
+                    && memcmp(&j->prefix, &addr, FAMILY_ADDRESS_SIZE(address->family)) == 0) {
```

Comment:
I may misunderstand the code, but I think the size for `memcmp()` should be `DIV_ROUND_UP(prefix_len, 8)` instead of `FAMILY_ADDRESS_SIZE(address->family)`.

Reply:
rest are zero bytes

Reply:
Is it assumed that the prefix length is fixed?

Reply:
yes

Reply:
Then the length should be 8.